### PR TITLE
Add language selection through config with whisper + Improve tests

### PR DESCRIPTION
**Pull request by AudranBert**
Add language selection for streaming with whisper, by default it will take the language found in the env settings. But you can pass a language in the config when starting streaming.

It also adds the possibility to pass a language in the config in case of offline decoding as requested in #53 . It will enable having a same model instance used for multiple languages instead of launching another Docker.

The PR is also improving tests to add tests about languages. Also removing some useless ones in order to reduce testing duration.

### CODE REVIEW

File: whisper/stt/processing/streaming.py [hunk 1]
```diff
@@ -43,17 +44,20 @@ async def wssDecode(ws: WebSocketServerProtocol, model_and_alignementmodel):
     try:
         config = json.loads(res)["config"]
         sample_rate = config["sample_rate"]
+        language = config.get("language", None)
```
Comment (Jeronymous) [inline whisper/stt/processing/streaming.py:47]: Don't we need to update the doc, specifying the language can be specified in the configuration of the streaming ?

Comment (Jeronymous) [reply whisper/stt/processing/streaming.py:47]: Also, why not to support language="`*`" (as in the env variable for the language in offline decoding)
that would need to replace "*" by None when reading the config

Comment (AudranBert) [reply whisper/stt/processing/streaming.py:47]: > Don't we need to update the doc, specifying the language can be specified in the configuration of the streaming request?

Yes I need to update the doc

**Comment by damienlaine** [comment]
Could you clarify the list of supported languages? For example, does it include "en," "fr," etc.? On the LinTO side, we consistently use BCP-47 codes for language representation.
Parsers (env, API directives...) shall at least support BCP-47 codes as inputs.

**Comment by Jeronymous** [comment]
> Could you clarify the list of supported languages? For example, does it include "en," "fr," etc.? On the LinTO side, we consistently use BCP-47 codes for language representation. Parsers (env, API directives...) shall at least support BCP-47 codes as inputs.

That did not changes in this PR.
several formats are supported : "fr" and "fr-FR". This holds for the whole LinTO speech toolkit.

Supported languages are listed here : https://github.com/linto-ai/linto-stt/blob/master/whisper/README.md#language

Also if the user gives a wrong one, it will give an explicit message with the list of possible ones (in the format "fr").

Why this question ? Do you think something is missing in the code or the documentation ?

**Comment by damienlaine** [comment]
I haven’t reviewed the code and relied on the doc:

- The docs mention "two or three-letter codes" for languages but not BCP-47 tags—should this be clarified?
- The PR focuses on streaming (?), but what about Celery (task) and HTTP service modes? Are specification updates planned for these?
- For Celery, should we open an issue in https://github.com/linto-ai/linto-transcription to handle the target language correctly?

**Comment by AudranBert** [comment]
>     * The PR focuses on streaming (?), but what about Celery (task) and HTTP service modes? Are specification updates planned for these?

The PR was created to fix the selection language in streaming, but I added the possibility to send the language through the config for streaming and offline (http and task).  That's why I linked this PR to the issue #53

**Comment by Jeronymous** [comment]
> * The docs mention "two or three-letter codes" for languages but not BCP-47 tags—should this be clarified?

Yes we should mention that they are supported, but that the second part ("FR" in "fr-FR") is ignored (results of the model are invariant to this)

> * The PR focuses on streaming (?), but what about Celery (task) and HTTP service modes? Are specification updates planned for these?

Yes. The PR is not finished yet (WIP)

> * For Celery, should we open an issue in https://github.com/linto-ai/linto-transcription to handle the target language correctly?

Yes. A feature request as an issue.
Worst case I will make it when I will commit related things
(our plan is to split : Audran here on core stt / me on transcription service API evolution)

File: whisper/stt/processing/utils.py [hunk 2]
```diff
@@ -55,8 +55,11 @@ def get_language(language = None):
     if language is None:
         language = os.environ.get("LANGUAGE", "*")
     # "fr-FR" -> "fr" (language-country code to ISO 639-1 code)
-    if len(language) > 2 and language[2] == "-":
-        language = language.split("-")[0]
+    language = language.split("-")[0].lower()
```
Comment (Jeronymous) [inline whisper/stt/processing/utils.py:58]: this line is wrong, remove it

File: whisper/stt/processing/utils.py [hunk 3]
```diff
@@ -65,11 +68,8 @@ def get_language(language = None):
         language = {v: k for k, v in LANGUAGES.items()}.get(language.lower(), language)
```
Comment (Jeronymous) [inline whisper/stt/processing/utils.py:68]: you see the lower here.
I think the one at line 62 is not needed.

File: whisper/stt/processing/utils.py [hunk 4]
```diff
@@ -65,11 +68,8 @@ def get_language(language = None):
...
         # Raise an exception for unknown languages
         if language not in LANGUAGES:
-            available_languages = (
-                list(LANGUAGES.keys())
-                + [k[0].upper() + k[1:] for k in LANGUAGES.values()]
```
Comment (Jeronymous) [inline whisper/stt/processing/utils.py:69]: Why did you remove capitalization ?
I think I did that for some reason.
Did you tested complete language names : how it is sensitive to the case ("French", "french", "FRENCH", ...) ?

File: whisper/stt/processing/utils.py (hunk 3 above)
Comment (AudranBert) [reply whisper/stt/processing/utils.py:68]: What is the purpose of that line, not sure to understand what it does?

File: whisper/stt/processing/utils.py (hunk 4 above)
Comment (AudranBert) [reply whisper/stt/processing/utils.py:69]: This `available_languages` only purpose is for the error message so it doesn't break anything to remove the capitalization.

File: whisper/stt/processing/utils.py (hunk 3 above)
Comment (Jeronymous) [reply whisper/stt/processing/utils.py:68]: I just mention that the lower was done somewhere else.
Read again my comment : look for "lower()" in the code, it appeas already where needed

Also, I think "French" (capitalized) is among the official list of languages by Whisper.
And we both agree that it is about have case insensitive input for the user.

File: whisper/stt/processing/utils.py (hunk 4 above)
Comment (Jeronymous) [reply whisper/stt/processing/utils.py:69]: OK, it will just print "french" instead of "French".
I think I wanted to differenciate the complete language names from codes ("fr") for more readability.
But I don't really care more than that.

File: whisper/stt/processing/utils.py (hunk 3 above)
Comment (AudranBert) [reply whisper/stt/processing/utils.py:68]: > you see the lower here. I think the one at line 62 is not needed.

I was just saying that the one that is not needed is the one at line 68 not the one at line 62. If we keep the one at line 68 and not the one at line 62 we can't use "FR" or "Fr" as language argument

**Comment by AudranBert** [comment]
Tests are running, I don't know how much time it will take to finish
